src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/service/RefDataService.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.iacasepaymentsapi.domain.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.iacasepaymentsapi.domain.entities.fee.OrganisationResponse;
+import uk.gov.hmcts.reform.iacasepaymentsapi.infrastructure.clients.IdamApi;
+import uk.gov.hmcts.reform.iacasepaymentsapi.infrastructure.clients.RefDataApi;
+import uk.gov.hmcts.reform.iacasepaymentsapi.infrastructure.clients.model.idam.UserInfo;
+import uk.gov.hmcts.reform.iacasepaymentsapi.infrastructure.security.RequestUserAccessTokenProvider;
+
+@Service
+@Slf4j
+public class RefDataService {
+
+    private final RefDataApi refDataApi;
+    private final IdamApi idamApi;
+    private final RequestUserAccessTokenProvider userAuthorizationProvider;
+    private final AuthTokenGenerator serviceAuthorizationProvider;
+
+    public RefDataService(
+        RefDataApi refDataApi,
+        RequestUserAccessTokenProvider userAuthorizationProvider,
+        AuthTokenGenerator serviceAuthorizationProvider,
+        IdamApi idamApi
+    ) {
+        this.refDataApi = refDataApi;
+        this.idamApi = idamApi;
+        this.userAuthorizationProvider = userAuthorizationProvider;
+        this.serviceAuthorizationProvider = serviceAuthorizationProvider;
+    }
+
+    private OrganisationResponse makeOrganisationRequest() {
+
+        String serviceToken = serviceAuthorizationProvider.generate();
+        UserInfo userInfo = idamApi.userInfo("Bearer " + serviceToken);
```

Comment:
You need to use user Token instead of service Token

Reply:
Done